test/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/claims/SelectDutiesControllerSpec.scala

```diff
@@ -59,34 +65,69 @@ class SelectDutiesControllerSpec
 
   implicit lazy val messages: Messages = MessagesImpl(Lang("en"), messagesApi)
 
-  private def sessionWithClaimState(
-    maybeDutiesSelectedAnswer: Option[DutiesSelectedAnswer]
-  ): (SessionData, FillingOutClaim, DraftC285Claim) = {
-    val draftC285Claim      =
-      DraftC285Claim.newDraftC285Claim.copy(dutiesSelectedAnswer = maybeDutiesSelectedAnswer)
+  def getMRNAnswer(): MovementReferenceNumberAnswer         = CompleteMovementReferenceNumberAnswer(Right(sample[MRN]))
+  def getEntryNumberAnswer(): MovementReferenceNumberAnswer = CompleteMovementReferenceNumberAnswer(
+    Left(sample[EntryNumber])
+  )
+
+  private def getSessionWithPreviousAnswer(
+    maybeDutiesSelectedAnswer: Option[DutiesSelectedAnswer],
+    movementReferenceNumberAnswer: MovementReferenceNumberAnswer,
+    displayDeclaration: Option[DisplayDeclaration] = None,
+    basisOfClaim: BasisOfClaimAnswer = CompleteBasisOfClaimAnswer(PersonalEffects)
+  ): (SessionData, FillingOutClaim) = {
+    val draftC285Claim      = DraftC285Claim.newDraftC285Claim.copy(
+      dutiesSelectedAnswer = maybeDutiesSelectedAnswer,
+      movementReferenceNumberAnswer = Some(movementReferenceNumberAnswer),
+      displayDeclaration = displayDeclaration,
+      basisOfClaimAnswer = Some(basisOfClaim)
+    )
```

Comment:
We need a separate ticket to move it in a single place. Maybe to put it in `DraftC285Claim` would do. And refactor all code where something like `getSessionWithPreviousAnswer` is used. No actions now. What do you think?

Reply:
These are very specific to a given tests.
I agree for a general case when you are only testing a controller which only depends on it's own data , we could create a general model. In this case however, this controller depends on responses given on 3 other pages